application-templates/starter/src/components/channel-details/channel-details-form.js

```diff
@@ -0,0 +1,100 @@
+import PropTypes from 'prop-types';
+import { useFormik } from 'formik';
+import { useIntl } from 'react-intl';
+import LocalizedTextField from '@commercetools-uikit/localized-text-field';
+import TextField from '@commercetools-uikit/text-field';
+import Spacings from '@commercetools-uikit/spacings';
+import SelectField from '@commercetools-uikit/select-field';
+import { CHANNEL_ROLES } from './constants';
+import validate from './validate';
+import messages from './messages';
+
+const getRoleOptions = Object.keys(CHANNEL_ROLES).map((key) => ({
+  label: CHANNEL_ROLES[key],
+  value: CHANNEL_ROLES[key],
+}));
+
+const ChannelDetailsForm = (props) => {
+  const intl = useIntl();
+  const formik = useFormik({
+    initialValues: props.initialValues,
+    onSubmit: props.onSubmit,
+    validate,
+    enableReinitialize: true,
+  });
+
+  const formElements = (
+    <Spacings.Stack scale="l">
+      <TextField
+        name="key"
+        title={intl.formatMessage(messages.channelKeyLabel)}
+        value={formik.values.key}
+        errors={formik.errors.key}
+        touched={formik.touched.key}
+        onChange={formik.handleChange}
+        onBlur={formik.handleBlur}
+        isReadOnly={props.isReadOnly}
+        renderError={(errorKey) => {
+          switch (errorKey) {
+            case 'duplicate':
+              return intl.formatMessage(messages.duplicateKey);
+            default:
```

Comment:
I was wondering here if it's correct to not render anything if the `errorKey` is not a expected one. It seems that, in this case, as error would have happened but wouldn't let the user know, right:

Reply:
Yes exactly. I mean we could return the error key or a more general message if we want to but usually these are known error keys from the validation.

I think with TS it would be easier to avoid forgetting to handle an error key.